cpmpy/expressions/globalconstraints.py

```diff
@@ -312,6 +312,146 @@ def value(self):
         return arrval in tab
 
 
+class MDD(GlobalConstraint):
+    """The values of the variables in 'array' correspond to a path in the mdd formed by the transition in 'transitions'
+    Root node is the first one used as a start in the first transition
```

Comment:
"first one used as a start" is confusing (are there multiple starts?) and doesn't convey that well that the first provided transition's initial state will be used as the "source" of the graph. Make this assumption more clear.

Reply:
add an example:
```
            0   ┌──┐ 2                                                    
         ┌─────►│n1├──┐                                                   
         │      └──┘  │  ┌──┐ 0                                           
         │            ├─►│n4├──┐                                          
        ┌┴┐ 1   ┌──┐ 2│  └──┘  │  ┌─┐                                     
        │r├────►│n2├──┘        ├─►│t│                                     
        └┬┘     └──┘     ┌──┐ 0│  └─┘                                     
         │            ┌─►│n5├──┘                                          
         │  2   ┌──┐ 0│  └──┘                                             
         └─────►│n3├──┘                                                   
                └──┘                                                      

    r = root node
    array = [r, n1, n2, n3, n4, n5n, t]
    transitions = [(r, 0, n1), (r, 1, n1), (r, 2, n3), (n1, 0, n4), (n2, 2, n4), (n3, 0, n5), (n4, 0, t), (n5, 0, t)]
```

Reply:
Describe the required structure of the "transition" argument

Reply:
Describe a bit more what MDD does by giving a simple example of a solution to the above graph.